src/server/pkg/work/work.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package work;
+option go_package = "github.com/pachyderm/pachyderm/src/server/pkg/work";
+
+// import "google/protobuf/any.proto";
+
+enum State {
+  RUNNING = 0;
+  COMPLETE = 1;
+  FAILED = 2;
+}
+
+message Work {
+  string id = 1;
+  // google.protobuf.Any meta = 2;
```

Comment:
What was/is the purpose of these `meta` fields?

Reply:
They are what I am addressing in the second part of the top comment for the PR. Using the Any protobuf type would allow the client to serialize additional arbitrary metadata that could be communicated between the master/worker (such as path ranges in the distributed merge case or chunk stats in the job processing case).

Reply:
Ahh, I see, an `Any` proto seems like a fine way to represent this. It assumes that workers will know how to deserialize said proto, which I think is reasonable. I think meta isn't a super descriptive name though. What about `Spec` since it's specifying what the task is?

Reply:
Those fields are used for both specifying the task and communicating results. `Spec` seems to be limited to just the former part. I am thinking of `Meta` as just general purpose metadata storage to be used by the client. The client can use it as just a spec when setting up the task in the master, or also update the fields in workers and collect the results in the master. I am not really against `Spec`, I just think it might suggest a more narrow scope than what that field can and should be used for.

Reply:
What if we just call it `Data` then? I'm not sure in what sense this data is meta, it seems like it's just the data associated with the job (or maybe pointers to it). But it feels weird to talk about meta data when there isn't any non-meta data visible.

Reply:
That makes sense.